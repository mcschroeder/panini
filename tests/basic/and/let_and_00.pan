and : (a:bool) -> (b:bool) -> {c:bool| c = true <=> (a = true /\ b = true)}

f00 = 
  let p1 = true in
  let p2 = true in
  and p1 p2

f01 = 
  let p1 = true in
  let p2 = false in
  and p1 p2

f02 = 
  let p1 = false in
  let p2 = true in
  and p1 p2

f03 = 
  let p1 = false in
  let p2 = false in
  and p1 p2


f10 : {z:bool|?}
f10 =  
  let p1 = true in
  let p2 = true in
  and p1 p2

f11 : {z:bool|?}
f11 =  
  let p1 = true in
  let p2 = false in
  and p1 p2

f12 : {z:bool|?}
f12 =  
  let p1 = false in
  let p2 = true in
  and p1 p2

f13 : {z:bool|?}
f13 =  
  let p1 = false in
  let p2 = false in
  and p1 p2


f20 : {z:bool|z=true}
f20 = 
  let p1 = true in
  let p2 = true in
  and p1 p2

f21 : {z:bool|z=false}
f21 = 
  let p1 = true in
  let p2 = false in
  and p1 p2

f22 : {z:bool|z=false}
f22 = 
  let p1 = false in
  let p2 = true in
  and p1 p2

f23 : {z:bool|z=false}
f23 = 
  let p1 = false in
  let p2 = false in
  and p1 p2
