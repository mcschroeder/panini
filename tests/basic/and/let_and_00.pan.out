f00 : {c:𝔹 | c = true}
f01 : {c:𝔹 | c = false}
f02 : {c:𝔹 | c = false}
f03 : {c:𝔹 | c = false}
f10 : {z:𝔹 | z = true}
f11 : {z:𝔹 | z = false}
f12 : {z:𝔹 | z = false}
f13 : {z:𝔹 | z = false}
f20 : {z:𝔹 | z = true}
f21 : {z:𝔹 | z = false}
f22 : {z:𝔹 | z = false}
f23 : {z:𝔹 | z = false}